infrastructure/database/repo/users.py

```diff
@@ -16,31 +18,57 @@ async def get_or_create_user(
     ):
         """
         Creates or updates a new user in the database and returns the user object.
+        Only fields provided as parameters will be updated in the case of a conflict.
         :param user_id: The user's ID.
         :param full_name: The user's full name.
         :param language: The user's language.
         :param username: The user's username. It's an optional parameter.
         :return: User object, None if there was an error while making a transaction.
         """
+        values_dict = {
+            "user_id": user_id,
+            "username": username,
+            "full_name": full_name,
+            "language": language,
 
-        insert_stmt = (
-            insert(User)
-            .values(
-                user_id=user_id,
-                username=username,
-                full_name=full_name,
-                language=language,
-            )
-            .on_conflict_do_update(
-                index_elements=[User.user_id],
-                set_=dict(
+        }
+
+        # Remove None values to avoid nullifying columns in the database
+        values_dict = {
+            key: value for key, value in values_dict.items() if value is not None
```

Comment:
There's no need to filter the values in this manner. Additionally, the `upsert` function below isn't as efficient as I initially believed, so I plan to refactor it. Please consider reverting all changes in this file.

Alternatively, you could rewrite the file to include a `select` function and, if no user is found, use `insert` to add them.